chore: Table Promotion from preview (#2600)
Workday/canvas-kit: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx

> the codemod, as its resulting formatting (spacing, quotes, etc.) may not match your project
> conventions.

## Component Promotions

### Table (Preview)

**PR:** [#2600](https://github.com/Workday/canvas-kit/pull/2600)

> We've promoted the `Table` component in preview to main.

Reviewer: ```suggestion
We've promoted the `Table` component in Preview(#preview) to Main(#main).
```

## Removals

### Table (Main)

**PR:** [#2600](https://github.com/Workday/canvas-kit/pull/2600)

> We've removed the `Table` component in main and replaced it with the `Table` component from preview.

Reviewer: ```suggestion
We've removed the `Table` component in Main(#main) and replaced it with the `Table` component from Preview(#preview).
```

This will consist of a new API since it is now a compound component.

> `rowState` prop is no longer a part of the `Table` component.

> See below for how these components will migrate:
> 
> |      v10       |         v11        |
> | -------------- | ------------------ |
> | `<Table/>`     | `<Table/>`         |
> |      `n/a`     | `<Table.Caption/>` |
> |      `n/a`     | `<Table.Head/>`    |
> |      `n/a`     | `<Table.Body/>`    |
> | `<TableRow/>`  | `<Table.Row/>`     |
> |      `n/a`     | `<Table.Header/>`  |
> |      `n/a`     | `<Table.Cell/>`    |
> |      `n/a`     | `<Table.Footer/>`  |

Reviewer 2: RC: Let's remove this. There's not enough correlation between the two tables for a comparison.
```suggestion
```

> In v10, `Table` and `TableRow` were the pieces that would be used to build a Table.
> In v11, `Table` is now a compound component that utilizes all the pieces of an html
> [Table](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table) element to build a Table.

Reviewer 2: ```suggestion
In v10, the main `Table` only exposed a `Table`  and `TableRow` component to build tables. In v11, `Table` now exposes every subcomponent of a semantic HTML
[Table](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table). While more verbose, it provides more flexibility for customization.
```

> 

Reviewer 2: Suggestion: Seems like we should have a basic example of the former Table and how to implement it in v11 with the new Table. Or we should link out to a GH discussion.

Please refer to the [Table examples](https://workday.github.io/canvas-kit/?path=/story/components-containers-table--basic) for how
to implement new tables.

> > **NOTE: This will have to be updated manually as the API is too different for a codemod.**
> 

Reviewer 2: RC: Let's remove this line. 
```suggestion
```

Reviewer: I think it's worth noting that if you're importing table from preview, make sure to change the import. There's not codemod for this since the change to compound component might be too complex, but worth notion about the import.

> the codemod, as its resulting formatting (spacing, quotes, etc.) may not match your project
> conventions.

## Component Promotions

### Table

**PR:** [#2600](https://github.com/Workday/canvas-kit/pull/2600)

We've promoted the `Table` compound component in [Preview](#preview) to [Main](#main). This replaces the existing `Table` component with a compound component
and will consist of a new compound component API.

> `rowState` prop is no longer a part of the `Table` component.

In v10, the `Table` component in [Main](#main) only exposed a `Table`  and `TableRow` component to build tables.

In v11, `Table` now exposes every subcomponent of a semantic HTML
[Table](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table). While more verbose, it provides more flexibility for customization.

See below for examples of the differences between the v10 `Table` and the v11 `Table` compound component.

v10:

> ```tsx

Reviewer: i would add the import to this example

<Table>
	<thead>
		<TableRow header={true}>
			<th>Header Text</th>
		</TableRow>
	</thead>
	<tbody>
		<TableRow>
			<td>Cell Text</td>
		</TableRow>
	</tbody>
</Table>
```

v11:

> ```tsx

Reviewer: same here, add the import

longer extends the `Box` component, however, it still supports Emotion `styled` and style props.
Type level components: `Title`, `Heading`, `BodyText` and `Subtext`, have not been changed since
they extend the `Text` component. These changes do not affect the components API.

## Glossary

> ### Main

Reviewer: can you add these to the top of the table of contents
